tests/fixtures/celery.py

```diff
@@ -0,0 +1,103 @@
+import os
+from collections.abc import Generator, Mapping, Sequence
+from dataclasses import dataclass
+from typing import Any
+from unittest.mock import PropertyMock, patch
+
+import pytest
+from celery import Celery
+from celery.worker import WorkController
+
+from core.celery.task import Task
+from core.service.celery.celery import task_queue_config
+from core.service.celery.configuration import CeleryConfiguration
+from core.service.celery.container import CeleryContainer
+from tests.fixtures.database import MockSessionMaker
+from tests.fixtures.services import ServicesFixture
+
+
+@pytest.fixture(scope="session")
+def celery_worker_parameters() -> Mapping[str, Any]:
+    """
+    Change the init parameters of Celery workers.
+
+    Normally when testing, we want to make sure that if there is an issue with the task
+    the worker will shut down after a certain amount of time. We default this to 30 sec.
+    However, when debugging it can be useful to set this to None, so you can set breakpoints
+    in the worker code, without the worker timing out and shutting down.
+    """
+    timeout = os.environ.get(
+        "PALACE_TEST_CELERY_WORKER_SHUTDOWN_TIMEOUT", "30.0"
+    ).lower()
+    shutdown_timeout = None if timeout == "none" or timeout == "" else float(timeout)
+    return {"shutdown_timeout": shutdown_timeout}
+
+
+@pytest.fixture(scope="session")
+def celery_pydantic_config() -> CeleryConfiguration:
+    """Configure the test Celery app.
+
+    The config returned will then be used to configure the `celery_app` fixture.
+    """
+    return CeleryConfiguration.construct(broker_url="memory://", cm_name="test")  # type: ignore[arg-type]
+
+
+@pytest.fixture(scope="session")
+def celery_config(celery_pydantic_config: CeleryConfiguration) -> Mapping[str, Any]:
+    """Configure the test Celery app.
+
+    The config returned will then be used to configure the `celery_app` fixture.
+    """
+    cm_name = celery_pydantic_config.cm_name
+    return celery_pydantic_config.dict() | task_queue_config(cm_name)
+
+
+@pytest.fixture(scope="session")
+def celery_parameters() -> Mapping[str, Any]:
+    """Change the init parameters of test Celery app.
+
+    The dict returned will be used as parameters when instantiating `~celery.Celery`.
+    """
+    return {"task_cls": "core.celery.task:Task"}
+
+
+@pytest.fixture(scope="session")
+def celery_includes() -> Sequence[str]:
+    """Include modules when a worker starts."""
+    return ("core.celery.worker",)
+
+
+@dataclass
+class CeleryFixture:
+    container: CeleryContainer
+    app: Celery
+    config: CeleryConfiguration
+    worker: WorkController
+
+
+@pytest.fixture()
+def celery_fixture(
+    services_fixture: ServicesFixture,
+    mock_session_maker: MockSessionMaker,
+    celery_app: Celery,
+    celery_worker: WorkController,
+    celery_pydantic_config: CeleryConfiguration,
+) -> Generator[CeleryFixture, None, None]:
```

Comment:
This is the main fixture for `Celery` that can be imported to be able to do a complete end to end run of a task inside a test. The task will be run in a worker thread.